workflow/rules/map.smk

```diff
@@ -505,6 +502,8 @@ rule convert_genome_to_sam_oligomap:
             config["cluster_log"], "oligomap_genome_to_sam_{sample}.log"
         ),
         nh=config["nh"],
+        out_dir=lambda wildcards, output: Path(output[0]).parent,
```

Comment:
I find this difficult to parse and accessing the `0` index possibly dangerous. What exactly happens here? And how does this connect to `output.gmap`? See comment below also (not clear to me why we don't still have just a single output file, much cleaner)

Reply:
If going back to the previous way of returning the output, this argument will be removed. Otherwise, using the new way of specifying the directories on the workflow, the value will be changed to the output directory directly.

Reply:
Yes, I think this will become much cleaner when you go back to writing to `STDOUT`.

Btw, changing an API is _not_ considered refactoring. Refactoring is when you change the internals of code:
- without changing the interface/API AND
- without changes to any outputs

You can make the code easier to read, create or remove classes and functions, improve the efficiency of the code etc., but from user's perspective, the usage and the outputs should be exactly the same.